store/tikv/region_cache.go

```diff
@@ -66,17 +79,57 @@ func (c *RPCContext) GetStoreID() uint64 {
 	return 0
 }
 
+// DropInvalidRegion returns true if the region is invalid and then dropped or if it has been dropped
+func (c *RegionCache) DropInvalidRegion(id RegionVerID) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	cachedregion, ok := c.mu.regions[id]
+	if !ok {
+		return true
+	}
+	// we need to check ttl because concurrent threads may fresh it
+	// For example, T1 find ttl valid but before T1 freshes the ttl, T2 find ttl invalid
+	// In this case, T2 will invoke this function but T1 may have freshed the ttl
+	// Then, T2 should not drop the cached region
+	lastAccess := atomic.LoadInt64(&cachedregion.lastAccess) // we may not need an atomic load, I think a simple load is enough?
+	lastAccessTime := time.Unix(lastAccess, 0)
+	if time.Since(lastAccessTime) < rcDefaultRegionCacheTTL {
+		return false
+	}
```

Comment:
Seems we can extract L94-L98 into a method of `CachedRegion`.

Reply:
Thanks, I will revise this.